packages/types/lib/statistics/IPartialResult.ts

```diff
@@ -0,0 +1,6 @@
+import type * as RDF from '@rdfjs/types';
+
+export interface IPartialResult {
+  data: RDF.Bindings | RDF.Quad;
```

Comment:
To enable more easy type-checking when using these partial results, I propose something like this:

```
export type PartialResult = {
  type: 'bindings';
  data: RDF.Bindings;
} | { type: 'quad'; ... };
```

Reply:
So I checked to see if this works, but the problem is that when you use such an approach you have to determine when your input is a `binding` or `quad`. I could make a mapping from operation to quad or binding but that seems like a lot of unnecessary work. How do you see this work?

Reply:
It can be used as follows:
```
const partialResult: PartialResult = ...;
if (partialResult.type === 'bindings') {
  // process data as bindings
}
```

Reply:
Or are you referring to using this interface in your transformer?

Perhaps we need different transform operations for the different types to simplify handling?

Reply:
I was referring to 'creating' the input to `updateStatistic`, but I see now that it shouldn't be too difficult.

Reply:
I've changed the typings around (a lot) following advice from Joachim. I'm not sure the typings are better this way so please let me know what you think!